core/tx_pool_test.go

```diff
@@ -400,6 +400,20 @@ func TestTransactionNegativeValue(t *testing.T) {
 	}
 }
 
+func TestTransactionZeroGasPrice(t *testing.T) {
+	t.Parallel()
+
+	pool, key := setupTxPool()
+	defer pool.Stop()
+
+	tx, _ := types.SignTx(types.NewTransaction(0, common.Address{}, big.NewInt(1), 200000, big.NewInt(0), nil, nil, nil, nil), types.HomesteadSigner{}, key)
```

Comment:
If I try to send this tx using an e2e test, I get an error saying "replay protection is required", should we be using the EIP155Signer here ?

Reply:
Probably. We can also use `LatestSigner` to automatically keep up to date with changes in the signer.